test/torch_np/numpy_tests/core/test_multiarray.py

```diff
@@ -697,7 +697,12 @@ def test_longdouble_assignment(self):
         # only relevant if longdouble is larger than float
         # we're looking for loss of precision
 
-        for dtype in (np.longdouble, np.longcomplex):
+        for dtype in (np.longdouble,) + (
+            # no complex256 in torch._numpy
+            (np.clongdouble,)
+            if hasattr(np, "clongdouble")
+            else ()
+        ):
```

Comment:
This probably shouldn't take 6 lines, maybe reformat a bit?

More importantly: how come this test passed before if complex long double isn't supported by `torch._numpy`?

Reply:
it doesn't :) this test is skipped precisely with the `reason="longdouble"` since extended floats and complex doesn't exist in `torch._numpy`:(https://github.com/pytorch/pytorch/blob/main/torch/_numpy/_dtypes.py#L174).

on second thought, this test (`test_longdouble_assignment) simply isn't compatible when using `torch._np` as `np`. Switched to conditionally skipping this test if `np = "torch._numpy` and removed `hasattr(np, "clongdouble")` check within the test body.